File: 2019/csharp/Program.cs
global using System.Text.RegularExpressions;
global using static System.Math;

using AdventOfCode2019;

var days = new IRobotElf[] {
            new Day1(),
        };

var day = args.Length > 0 && int.TryParse(args[0], out var temp) ? temp : days.Length;
day--;


Console.WriteLine($"Part1: {days[day].Part1()}");
Console.WriteLine($"Part2: {days[day].Part2()}");

File: 2019/fsharp/Program.fs
open AdventOfCode.FSharp.NorthPole
open AdventOfCode.FSharp.Y2019

let inline E v = v |> string |> Some

let inline old (part1: string -> string) (part2: string -> string) =
    [ fun data output ->
          let text = AdventOfCode.FSharp.Util.text data
          part1 text |> output 1
          part2 text |> output 2 ]

let days =
    [ 01, old Day01.part1 Day01.part2 ]
    |> List.map (fun (d, r) -> { year = 2019; day = d; runs = r })

runCommandLine days
